### PR TITLE
deprecation(semver): deprecate `==`, `===`, `!==`, and `""` operators

### PR DESCRIPTION
**Pull request by timreichen**
- deprecates `"=="`, `"==="`, `"!=="` operators, as they function the same as `"="` and `"!="` respectively. 
- makes `Comparator.operator` optional to replace `""` operator.

### CODE REVIEW

**Comment by kt3k** [comment]
> removes "==", "===", "!==" operators, as they function the same as "=" and "!=" respectively.
> makes Comparator.operator optional to replace "" operator.

It looks like the both are breaking changes

**Comment by iuioiua** [comment]
We should consider that stricter `==`, `===` and `!==` may exist because they are familiar to JavaScript devs. That being said, that may not justify having them in the API. I'm impartial.

**Comment by kt3k** [comment]
> Would you agree though that these operators do not make sense and should be removed?

Not sure it's worth the breaking change.

Also not sure we are strongly in favor of removing `''` from operator kind. I guess nobody cares about it.

Also we don't do breaking change without deprecation period.

**Review by kt3k — APPROVED** [review]
This probably makes sense. I've never seen semver range expression using these operators. e.g. No `==1.0.0` or `===1.0.0`, it's always `=1.0.0` in documents out there.

LGTM.

**Comment by iuioiua** [comment]
> * makes `Comparator.operator` optional to replace `""` operator.

Neither an optional `operator` nor `operator = ""` make sense to me. If `comparator.operator` is either `undefined` or `""`, it seizes to be a comparator and really is a `SemVer` at that point. Am I missing something?

**Comment by kt3k** [comment]
> If comparator.operator is either undefined or "", it seizes to be a comparator and really is a SemVer at that point. Am I missing something?

I think that's still a valid comparator because the expression like "1.0.0" is a valid range specifier, which matches the exact version "1.0.0"

**Comment by iuioiua** [comment]
A range that lacks an operator is surprising behaviour to me. It's also an exception to the rule: a comparator is a semver prefixed with an operator. Either way, it's a discussion for a separate issue. I won't block this PR.

**Comment by kt3k** [comment]
A range without operator is just fine in npm. See https://docs.npmjs.com/cli/v10/configuring-npm/package-json#dependencies

I think we shouldn't change that convention here

File: semver/_constants.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 
+/**
+ * @deprecated (will be removed in 0.217.0) `"=="`, `"==="`, `"!=="` and `""` operators are deprecated. Use `"="`, `"!="` or `undefined` instead.
```
Comment (kt3k) [inline semver/_constants.ts:4]:
```suggestion
 * @deprecated (will be removed in 0.219.0) `"=="`, `"==="`, `"!=="` and `""` operators are deprecated. Use `"="`, `"!="` or `undefined` instead.
```

File: semver/types.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export type ReleaseType =
 
 /**
  * SemVer comparison operators.
+ * @deprecated (will be removed in 0.217.0) `"=="`, `"==="`, `"!=="` and `""` operators are deprecated. Use `"="`, `"!="` or `undefined` instead.
```
Comment (kt3k) [inline semver/types.ts:21]:
```suggestion
 * @deprecated (will be removed in 0.219.0) `"=="`, `"==="`, `"!=="` and `""` operators are deprecated. Use `"="`, `"!="` or `undefined` instead.
```

**Pull request merged**